lib/ansible/modules/remote_management/lxca/lxca_nodes.py

```diff
@@ -0,0 +1,256 @@
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'supported_by': 'community',
+    'status': ['preview']
+}
+
+
+DOCUMENTATION = '''
+---
+version_added: "2.8"
+author:
+  - Naval Patel (@navalkp)
+  - Prashant Bhosale (@prabhosa)
+module: lxca_nodes
+short_description: Custom module for lxca nodes inventory utility
+description:
+  - This module returns/displays a inventory details of nodes
+
+options:
+  uuid:
+    description:
+      uuid of device, this is string with length greater than 16.
+
+  command_options:
+    description:
+      options to filter nodes information
+    default: nodes
+    choices:
+        - nodes
+        - nodes_by_uuid
+        - nodes_by_chassis_uuid
+        - nodes_status_managed
+        - nodes_status_unmanaged
+
+  chassis:
+    description:
+      uuid of chassis, this is string with length greater than 16.
+
+extends_documentation_fragment:
+    - lxca_common
+'''
+
+EXAMPLES = '''
+# get all nodes info
+- name: get nodes data from LXCA
+  lxca_nodes:
+    login_user: USERID
+    login_password: Password
+    auth_url: "https://10.243.15.168"
+    command_options: nodes
+
+# get specific nodes info by uuid
+- name: get nodes data from LXCA
+  lxca_nodes:
+    login_user: USERID
+    login_password: Password
+    auth_url: "https://10.243.15.168"
+    uuid: "3C737AA5E31640CE949B10C129A8B01F"
+    command_options: nodes_by_uuid
+
+# get specific nodes info by chassis uuid
+- name: get nodes data from LXCA
+  lxca_nodes:
+    login_user: USERID
+    login_password: Password
+    auth_url: "https://10.243.15.168"
+    chassis: "3C737AA5E31640CE949B10C129A8B01F"
+    command_options: nodes_by_chassis_uuid
+
+# get managed nodes
+- name: get nodes data from LXCA
+  lxca_nodes:
+    login_user: USERID
+    login_password: Password
+    auth_url: "https://10.243.15.168"
+    command_options: nodes_status_managed
+
+# get unmanaged nodes
+- name: get nodes data from LXCA
+  lxca_nodes:
+    login_user: USERID
+    login_password: Password
+    auth_url: "https://10.243.15.168"
+    command_options: nodes_status_unmanaged
+
+'''
+
+RETURN = r'''
+result:
+    description: nodes detail from lxca
+    returned: always
+    type: dict
+    sample:
+      nodeList:
+        - machineType: '6241'
+          model: 'AC1'
+          type: 'Rack-TowerServer'
+          uuid: '118D2C88C8FD11E4947B6EAE8B4BDCDF'
+          # bunch of properties
+        - machineType: '8871'
+          model: 'AC1'
+          type: 'Rack-TowerServer'
+          uuid: '223D2C88C8FD11E4947B6EAE8B4BDCDF'
+          # bunch of properties
+        # Multiple nodes details
+'''
+
+import traceback
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.remote_management.lxca.common import LXCA_COMMON_ARGS, has_pylxca
+try:
+    from pylxca import connect
+    from pylxca import disconnect
+    from pylxca import nodes
+except ImportError:
+    pass
```

Comment:
Isn't it better to  consider 'ImportError'  exception for an early failure decision before further function calls?

Reply:
Which function calls ? `has_pylxca`  handles case when `pylxca` is missing.

Reply:
In case of any exception `from pylxca import connect` or other imports here (without finding respective imported functions like `connect` or `disconnect`), it fails later again at  `setup_conn` and other actual method calls. Isn't it good to handle early?

Reply:
has_pylxca is called from main before it uses connect and disconnect. so I think I am safe there.

Reply:
In order to simplify the code, `main` could call `execute_module` directly (and then `run_tasks` method should be removed), apart from that module behavior seems fine.

Reply:
removed run_tasks and moved connection_object to module_utils.